packages/query/src/index.ts

```diff
@@ -721,10 +724,66 @@ const getQueryFnArguments = ({
   return '{ signal }';
 };
 
+const generateQueryKeyImplementation = ({
```

Comment:
By extracting the function, the distance between the declaration and the place where it is actually used has increased. This increases the burden of understanding the overall processing, so consider reverting it if there is no significant benefit.